core/iwasm/interpreter/wasm_loader.c

```diff
@@ -4054,17 +4062,48 @@ load_global_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
     return false;
 }
 
+static bool
+check_duplicate_exports(WASMModule *module, char *error_buf,
+                        uint32 error_buf_size)
+{
+    uint32 i;
+    char **names = loader_malloc(module->export_count * sizeof(char *),
+                                 error_buf, error_buf_size);
```

Comment:
If to reduce the chance of calling loader_malloc which may spend more time and be easier to get failed, we can define a bufer and only allocate memroy when the buffer isn't large enough.
```C
    char *names_buf[32], **names = names_buf;

    if (module->export_count > 32) {
        names = loader_malloc(...);
        ...
    }
    ...
```

Reply:
Done

Reply:
Took care of both comments, thank you!